File: absurd.c
#include<stdio.h>
long int removezero(long int c);
int main()
{
	long int t,c,i = 0,e1,e2,e3;
	long double ulimit,llimit;
	scanf("%ld",&t);
	while(i < t)
	{
		i ++;
		scanf("%ld",&c);
		c = removezero(c);
		llimit = 0.95*c;
		ulimit = 1.05*c;
		e1 = c - (c%10);
		if (e1 >= llimit && e1 <= ulimit)
		{
			printf("absurd\n");
			continue;
		}
		e2 = e1 + 5;
		if (e2 >= llimit && e2 <= ulimit && e2 != c)
		{
			printf("absurd\n");
			continue;
		}
		e3 = e2 + 5;
		if (e3 >= llimit && e3 <= ulimit)
		{
			printf("absurd\n");
			continue;
		}
		else
			printf("not absurd\n");
	}
	return 0;
}
long int removezero(long int c)
{
	if (c%10)
		return c;
	else
		return removezero(c/10);
}

File: absurd.py
def main ():
	t = input()
	i = 0
	while i < t:
		i += 1
		c = int(raw_input().strip('0'))
		llimit = .95*c
		ulimit = 1.05*c
		e1 = c - (c%10)
		if e1 >= llimit and e1 <= ulimit:
			# print "e1",e1
			print "absurd"
			continue
		e2 = e1 + 5
		if e2 >= llimit and e2 <= ulimit and e2 != c:
			# print "e2",e2
			print "absurd"
			continue
		e3 = e2 + 5
		if e3 >= llimit and e3 <= ulimit and e3 != c:
			# print "e3",e3
			print "absurd"
			continue
		else:
			print "not absurd"

if __name__ == '__main__':
	main()
